inference-engine/src/plaidml_plugin/ops/bucketize.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset3.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;  // NOLINT[build/namespaces]
+using ngraph::opset3::Bucketize;
+
+namespace PlaidMLPlugin {
+
+static OpRegistration reg("Bucketize", [](const Context& ctx) {
+  auto* layer = ngraph::as_type<Bucketize>(ctx.layer);
+  IE_ASSERT(ctx.operands.size() == 2);
+  auto A = ctx.operands.at(0);
+  auto B = ctx.operands.at(1);
+
+  std::vector<int> broadcastShape;
+  std::vector<int> bcast_axes;
+  auto A_shape = A.compute_shape().sizes();
+  int count = A_shape.size();
+  for (int i = 0; i < count; i++) {
+    // broadcast requires shape with int type
+    broadcastShape.push_back(static_cast<int>(A_shape[i]));
+    bcast_axes.push_back(i);
+  }
+  // Bucket is a 1-D tensor, the first dimension shall be tensor size
+  auto B_shape = B.compute_shape().sizes();
+  IE_ASSERT(B_shape.size() == 1);
+  broadcastShape.push_back(static_cast<int>(B_shape[0]));
+  auto broadcastResult = op::broadcast(A, broadcastShape, bcast_axes);
```

Comment:
The `compute_shape` function is extremely costly in performance, and it should be avoided wherever possible. When broadcasting, we can do this by using implicit broadcast instead of explicit broadcast. (In other situations we could use `bind_dims`.)

In this case, I believe we can use unsqueeze & repeat to produce `broadcastResult`:
```
IE_ASSERT(B.rank() == 1);
std::vector<TensorDim> B_dims(B.rank());
B.bind_dims(B_dims)
broadcastResult = op::repeat(op::unsqueeze(A, {-1}), B_dims[0], -1);
```

This requires a change to `op::repeat` (which currently only takes integers, but can easily be rewritten to accept `TensorDim`s as well). I am working on that change in https://github.com/plaidml/plaidml/pull/1518

Reply:
n.b. the change in plaidml/plaidml#1518 has landed

Reply:
Need to add new version of repeat op in op.h,  changed code can work with https://github.com/plaidml/plaidml/pull/1522, but not sure if we shall change all repeat op to use new type of argument.